e2e-tests/playwright/tests/functional/channels/settings/display/theme_accessibility.spec.ts

```diff
@@ -0,0 +1,115 @@
+
+    // * The theme section for premade themes should have no accessibility violations
+    // accessibilityScanResults = await ab.analyze()
+    // expect(accessibilityScanResults.violations).toHaveLength(0);
+
+    // * Should be able to tab through the options
+    await page.keyboard.press('Tab');
+    await expect(page.getByRole('button', {name: 'Denim'})).toBeFocused();
+    await page.keyboard.press('Tab');
+    await expect(page.getByRole('button', {name: 'Sapphire'})).toBeFocused();
+    await page.keyboard.press('Tab');
+    await expect(page.getByRole('button', {name: 'Quartz'})).toBeFocused();
+    await page.keyboard.press('Tab');
+    await expect(page.getByRole('button', {name: 'Indigo'})).toBeFocused();
+    await page.keyboard.press('Tab');
+    await expect(page.getByRole('button', {name: 'Onyx'})).toBeFocused();
+    await page.keyboard.press('Tab');
+
+    // Note: There's no "Apply to all your teams" option because this user is only on one team
+    await expect(page.getByRole('link', {name: 'See other themes'})).toBeFocused();
+    await page.keyboard.press('Tab');
+    await expect(page.getByRole('button', {name: 'Save'})).toBeFocused();
+
+    // # Go back to the premade/custom option
+    await page.getByLabel('Premade Themes').focus();
+
+    // # Switch to the custom theme section
+    await page.keyboard.press('ArrowDown');
+
+    // * The Custom Theme option should be focused
+    await expect(page.getByLabel('Custom Theme')).toBeFocused();
+
+    // * Check the ARIA of the Sidebar Styles section
+    await page.keyboard.press('Tab');
+    const sidebarStyles = page.getByRole('button', {name: 'Sidebar Styles'});
+    await expect(sidebarStyles).toBeFocused();
+    await expect(sidebarStyles).toHaveAttribute('aria-expanded', 'false');
+
+    // * Check that we can tab over the collapsed section to the next section and then back again
+    await page.keyboard.press('Tab');
+    await expect(page.getByRole('button', {name: 'Center Channel Styles'})).toBeFocused();
+
+    await page.keyboard.press('Shift+Tab');
+    await expect(sidebarStyles).toBeFocused();
+
+    // * Check that we can expand the section
+    await page.keyboard.press('Enter');
+    await expect(sidebarStyles).toHaveAttribute('aria-expanded', 'true');
+
+    // # Wait for the expanding animation to be open
+    await expect(page.getByLabel('Sidebar BG', {exact: true})).toBeVisible();
+
+    // * Check that we can tab through color pickers
+    await page.keyboard.press('Tab');
+    await expect(page.getByLabel('Sidebar BG', {exact: true})).toBeFocused();
+    await page.keyboard.press('Tab');
+    await expect(page.getByLabel('Sidebar Text', {exact: true})).toBeFocused();
+    await page.keyboard.press('Tab');
+    await expect(page.getByLabel('Sidebar Header BG')).toBeFocused();
+
+    // * The theme section for custom themes should have no accessibility violations
+    // accessibilityScanResults = await ab.analyze()
```

Comment:
is this expected to be commented? there are other similar in this file. I am not that familiar with playwright.

Reply:
Oh yeah, I forgot to mention that I intentionally commented those out because of other a11y issues in the settings modal.

There might've been a contrast issue, but the problem I remember is that Axe doesn't like the way that we used ARIA to group items in the settings sidebar. There's no standard pattern for grouping tabs together, and apparently the way we tried to do that is wrong. I made a note to look into that more when we get a chance

Reply:
One option to incrementally run Accessibility tests on a page is by using flags like `disableRules`, `include` and `exclude` (see https://www.npmjs.com/package/@axe-core/playwright). This way, we can protect specific sections/components with Accessibility in place, then gradually lift the flags until the entire page is covered. Also, I'd recommend adding a comment on the related ticket to track any ongoing work with these flags.

Reply:
Great idea. I just went into this and found that one of the failures is actually something added in https://github.com/mattermost/mattermost/pull/29596 which we did wrong (having `aria-controls` on the Premade Themes/Custom Themes inputs), so turning that on to ensure we don't do anything new wrong is a great idea